File: packages/core/tests/fixtures/code-nav/sample.java
package com.example.auth;

import java.util.Map;
import java.util.HashMap;

public class AuthMiddleware {
    private final AuthConfig config;
    private final Map<String, String> cache;

    public AuthMiddleware(AuthConfig config) {
        this.config = config;
        this.cache = new HashMap<>();
    }

    public String authenticate(String token) throws Exception {
        if (token == null || token.isEmpty()) {
            throw new Exception("no token");
        }
        return "user-1";
    }

    public String refreshToken(String token) {
        return token + "-refreshed";
    }

    private boolean validateJWT(String jwt) {
        return jwt != null && !jwt.isEmpty();
    }
}

interface Authenticator {
    boolean verify(String token);
}

enum UserRole {
    ADMIN,
    USER,
    GUEST
}

File: packages/core/tests/fixtures/code-nav/sample.rs
use std::collections::HashMap;
use crate::config::Settings;

pub struct AuthConfig {
    pub secret: String,
    pub expires_in: u64,
}

pub struct AuthMiddleware {
    config: AuthConfig,
    cache: HashMap<String, String>,
}

impl AuthMiddleware {
    pub fn new(config: AuthConfig) -> Self {
        AuthMiddleware {
            config,
            cache: HashMap::new(),
        }
    }

    pub fn authenticate(&self, token: &str) -> Result<String, String> {
        if token.is_empty() {
            return Err("no token".to_string());
        }
        Ok("user-1".to_string())
    }

    fn validate_jwt(&self, jwt: &str) -> bool {
        !jwt.is_empty()
    }
}

pub trait Authenticator {
    fn verify(&self, token: &str) -> bool;
}

pub fn create_middleware(config: AuthConfig) -> AuthMiddleware {
    AuthMiddleware::new(config)
}

pub const DEFAULT_SECRET: &str = "dev";
